### PR TITLE
Issue #SB-17038 fix:Portal - User provides structured feedback on content

### PR DESCRIPTION
**Pull request by ArjunKhetia-NIIT**

### CODE REVIEW

File: src/app/client/src/app/modules/player-helper/components/content-rating/content-rating.component.ts [hunk 1]
```diff
@@ -35,20 +42,55 @@ export class ContentRatingComponent implements OnInit, OnDestroy {
...
+    });
+  }
+
+  public changeOptions(options) {
+    if (options['key'] === 'OTHER' && !this.showTextarea) {
+      this.showTextarea = true;
+    } else if (options['key'] === 'OTHER' && this.showTextarea) {
+      this.showTextarea = false;
+    }
+    if (this.comments.indexOf(options['key']) === -1) {
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/player-helper/components/content-rating/content-rating.component.ts]: _.contains

File: src/app/client/src/app/modules/player-helper/components/content-rating/content-rating.component.ts [hunk 2]
```diff
@@ -35,20 +42,55 @@ export class ContentRatingComponent implements OnInit, OnDestroy {
...
+ constructor(resourceService: ResourceService, toasterService: ToasterService, public formService: FormService,
+    telemetryService: TelemetryService, activatedRoute: ActivatedRoute) {
     this.resourceService = resourceService;
     this.toasterService = toasterService;
     this.telemetryService = telemetryService;
     this.activatedRoute = activatedRoute;
   }
 
   ngOnInit() {
+    this.startext = this.resourceService.frmelmnts.lbl.defaultstar;
```
[the remark this answers is not in the document]
Comment (ArjunKhetia-NIIT) [reply src/app/client/src/app/modules/player-helper/components/content-rating/content-rating.component.ts]: done

File: src/app/client/src/app/modules/player-helper/components/content-rating/content-rating.component.ts [hunk 3]
```diff
@@ -35,20 +42,55 @@ export class ContentRatingComponent implements OnInit, OnDestroy {
...
+    this.resourceService.languageSelected$.subscribe(item => {
+      const formReadInputParams = {
+        formType: 'contentfeedback',
+        contentType: item.value,
+        formAction: 'get'
+      };
+      this.formService.getFormConfig(formReadInputParams).subscribe(
```
[the remark this answers is not in the document]
Comment (ArjunKhetia-NIIT) [reply src/app/client/src/app/modules/player-helper/components/content-rating/content-rating.component.ts]: done

File: src/app/client/src/app/modules/player-helper/components/content-rating/content-rating.component.ts [hunk 4]
```diff
@@ -35,20 +42,55 @@ export class ContentRatingComponent implements OnInit, OnDestroy {
...
+        (formResponsedata) => {
+          const resObj = formResponsedata[0];
+          this.feedbackObj = [resObj[1], resObj[2], resObj[3], resObj[4], resObj[5]];
```
[the remark this answers is not in the document]
Comment (ArjunKhetia-NIIT) [reply src/app/client/src/app/modules/player-helper/components/content-rating/content-rating.component.ts]: done

File: src/app/client/src/app/modules/player-helper/components/content-rating/content-rating.component.ts (hunk 1 above)
Comment (ArjunKhetia-NIIT) [reply src/app/client/src/app/modules/player-helper/components/content-rating/content-rating.component.ts]: done

**Pull request closed without merging**